src/jit/codegencommon.cpp

```diff
@@ -4613,17 +4616,23 @@ void CodeGen::genCheckUseBlockInit()
         }
     }
 
-    // After debugging this further it was found that this logic is incorrect:
-    // it incorrectly assumes the stack slots are always 4 bytes (not necessarily the case)
-    // and this also double counts variables (we saw this in the debugger) around line 4829.
-    // Even though this doesn't pose a problem with correctness it will improperly decide to
-    // zero init the stack using a block operation instead of a 'case by case' basis.
+    // Record number of 4 byte slots that need zeroing.
     genInitStkLclCnt = initStkLclCnt;
 
-/* If we have more than 4 untracked locals, use block initialization */
-/* TODO-Review: If we have large structs, bias toward not using block initialization since
-   we waste all the other slots.  Really need to compute the correct
-   and compare that against zeroing the slots individually */
+    // Decide if we will do block initialization in the prolog, or use
+    // a series of individual stores.
+    //
+    // Primary factor is the number of slots that need zeroing. We've
+    // been counting by sizeof(int) above. We assume for now we can
+    // only zero register width bytes per store.
```

Comment:
Is this `integer` register width bytes per store (i.e. `32-bit` or `64-bit`) or does it also take into account other registers (such as `xmm` which is `128-bit` or `ymm` which is `256-bit`)?

Reply:
Currently just integer registers.

xmm would likely offer further improvement, especially if there's no need for a cleanup store. But that would be a somewhat more involved change, and we're trying hard to close down jit work for 3.0. And if we get to that point, we should also think more about the strategy here and if or how we could share logic or code with other block inits we do.

For ymm, potential is less clear; the stack is only guaranteed to be 16 byte aligned.

Reply:
Might be worth logging a tracking issue for future investigation.

I was mostly just wondering since this just says "register width" and the clarification is useful 😄

Reply:
Think we can use #19076 to track further improvements.

Reply:
Also perhaps #13827.